PROJECT_MCPU/MCPU/DOC/GENERATED/man/man3/Notify.3
.TH "Notify" 3 "MCPU" \" -*- nroff -*-
.ad l
.nh
.SH NAME
Notify \- This is the calss implementing the application message management\&.  

.SH SYNOPSIS
.br
.PP
.PP
\fR#include <Notify\&.h>\fP
.SS "Classes"

.in +1c
.ti -1c
.RI "class \fBitem\fP"
.br
.ti -1c
.RI "class \fBtranslate\fP"
.br
.in -1c
.SS "Public Types"

.in +1c
.ti -1c
.RI "enum class \fBmessages\fP { \fBNO_MESSAGE\fP =0, \fBERROR_FATAL_COMPILATION\fP, \fBERROR_AWS_INITIALIZATION\fP, \fBERROR_PCB301_RESET\fP, \fBERROR_PCB302_RESET\fP, \fBERROR_PCB303_RESET\fP, \fBERROR_PCB304_RESET\fP, \fBERROR_PCB325_RESET\fP, \fBERROR_PCB326_RESET\fP, \fBERROR_PCB301_COMMUNICATION_ERROR\fP, \fBERROR_PCB302_COMMUNICATION_ERROR\fP, \fBERROR_PCB303_COMMUNICATION_ERROR\fP, \fBERROR_PCB304_COMMUNICATION_ERROR\fP, \fBERROR_PCB325_COMMUNICATION_ERROR\fP, \fBERROR_PCB326_COMMUNICATION_ERROR\fP, \fBERROR_XRAY_PUSH_FAULT\fP, \fBERROR_MOTOR_PEDALS_UD_FAULT\fP, \fBERROR_MOTOR_PEDALS_COMP_FAULT\fP, \fBERROR_MOTOR_BUTTON_CARM_FAULT\fP, \fBERROR_MOTOR_BUTTON_UD_FAULT\fP, \fBERROR_CONFIGURATION_FAULT\fP, \fBERROR_POWER_DOWN_ERROR\fP, \fBERROR_EMERGENCY_BUTTON\fP, \fBERROR_CABINET_SAFETY\fP, \fBERROR_BATTERY_LOW_ERROR\fP, \fBERROR_GENERATOR_ERROR_CONNECTION\fP, \fBERROR_COLLIMATION_SELECTION_ERROR\fP, \fBERROR_FILTER_SELECTION_ERROR\fP, \fBERROR_MIRROR_SELECTION_ERROR\fP, \fBERROR_STATOR_SENSOR_LOW\fP, \fBERROR_STATOR_SENSOR_SHORT\fP, \fBERROR_BULB_SENSOR_LOW\fP, \fBERROR_BULB_SENSOR_SHORT\fP, \fBERROR_BODY_MOTOR_BRAKE_FAULT\fP, \fBERROR_BODY_POTENTIOMETER_READ_FAULT\fP, \fBERROR_TILT_MOTOR_BRAKE_FAULT\fP, \fBERROR_TILT_MOTOR_HOMING\fP, \fBERROR_VERTICAL_MOTOR_HOMING\fP, \fBERROR_VERTICAL_LIMIT_SWITCH\fP, \fBERROR_ARM_MOTOR_HOMING\fP, \fBERROR_SLIDE_MOTOR_HOMING\fP, \fBERROR_BODY_MOTOR_HOMING\fP, \fBERROR_BODY_LIMIT_SWITCH\fP, \fBERROR_GENERATOR_SETUP\fP, \fBERROR_GENERATOR_SYS_MESSAGE\fP, \fBERROR_POTTER_GRID_POSITION\fP, \fBERROR_X_RAY_BUTTON_RELEASED\fP, \fBERROR_X_RAY_LOW_ANODIC\fP, \fBERROR_X_RAY_LOW_KV\fP, \fBERROR_X_RAY_TIMEOUT\fP, \fBERROR_X_RAY_STARTER\fP, \fBWARNING_MESSAGES_SECTION\fP, \fBWARNING_MISSING_COMPRESSION\fP, \fBWARNING_MISSING_PATIENT_PROTECTION\fP, \fBWARNING_ARM_POSITION_WARNING\fP, \fBWARNING_POSITION_NOT_CALIBRATED\fP, \fBWARNING_FORCE_NOT_CALIBRATED\fP, \fBWARNING_MISSING_PADDLE\fP, \fBWARNING_WRONG_PADDLE\fP, \fBWARNING_MISSING_EXPOSURE_MODE\fP, \fBWARNING_MISSING_EXPOSURE_DATA\fP, \fBWARNING_XRAY_BUTTON_DISABLED\fP, \fBWARNING_DOOR_STUDY_OPEN\fP, \fBWARNING_GENERATOR_NOT_READY\fP, \fBWARNING_TUBE_TEMPERATURE\fP, \fBWARNING_COLLI_LIGHT_ACTIVATION\fP, \fBWARNING_STATOR_SENSOR_HIGH\fP, \fBWARNING_BULB_SENSOR_HIGH\fP, \fBWARNING_ANODE_TEMPERATURE_HIGH\fP, \fBWARNING_FILTER_OUT_OF_POSITION\fP, \fBWARNING_MIRROR_OUT_OF_POSITION\fP, \fBWARNING_COLLIMATOR_OUT_OF_POSITION\fP, \fBWARNING_GENERATOR_SERVICE_MODE\fP, \fBWARNING_GRID_OUT_OF_POSITION\fP, \fBWARNING_GRID_GENERAL_ENABLE\fP, \fBINFO_MESSAGES_SECTION\fP, \fBINFO_BATTERY_DISABLED\fP, \fBINFO_PROJECTION_CONFIRMATION\fP, \fBINFO_PROJECTION_ABORT\fP, \fBINFO_SLIDE_ACTIVATION_CONFIRMATION\fP, \fBINFO_BODY_ACTIVATION_CONFIRMATION\fP, \fBINFO_ARM_ACTIVATION_CONFIRMATION\fP, \fBINFO_TILT_ACTIVATION_CONFIRMATION\fP, \fBINFO_SLIDE_ACTIVATION_FAILED\fP, \fBINFO_BODY_ACTIVATION_FAILED\fP, \fBINFO_ARM_ACTIVATION_FAILED\fP, \fBINFO_TILT_ACTIVATION_FAILED\fP, \fBINFO_BODY_DRIVER\fP, \fBINFO_VERTICAL_DRIVER\fP, \fBINFO_TILT_DRIVER\fP, \fBINFO_ARM_DRIVER\fP, \fBINFO_SLIDE_DRIVER\fP, \fBINFO_MOTOR_48V_SAFETY_LINE_OFF\fP, \fBINFO_MOTOR_48V_POWER_SUPPLY_OFF\fP, \fBINFO_BURNING_JUMPER_PRESENT\fP, \fBINFO_POWER_LOCK\fP, \fBINFO_ACTIVATION_MOTOR_MANUAL_DISABLE\fP, \fBINFO_ACTIVATION_MOTOR_SAFETY_DISABLE\fP, \fBINFO_ACTIVATION_MOTOR_ERROR_DISABLE\fP, \fBINFO_POWER_OFF_REQUEST_ACTIVATION\fP, \fBINFO_GRID_INOUT_MANUAL_MODE\fP, \fBLABEL_MESSAGES_SECTION\fP, \fBLABEL_ERROR_WINDOW_PANEL\fP, \fBLABEL_PROJECTION_SELECTION_PANEL\fP, \fBLABEL_ZERO_SETTING_PANEL_TITLE\fP, \fBLABEL_ROTATION_TOOL_PANEL_TITLE\fP, \fBLABEL_SERVICE_PANEL_TITLE\fP, \fBLABEL_CALIBRATION_PANEL_TITLE\fP, \fBLABEL_CALIBRATION_XRAY_TUBE_TITLE\fP, \fBLABEL_NOT_READY_FOR_EXPOSURE\fP, \fBLABEL_READY_FOR_EXPOSURE\fP, \fBLABEL_X_RAY_ON\fP, \fBLABEL_MOTOR_COMMAND_SUCCESS\fP, \fBLABEL_MOTOR_COMMAND_MANUAL_TERMINATION\fP, \fBLABEL_MOTOR_ERROR_OBSTACLE_DETECTED\fP, \fBLABEL_MOTOR_ERROR_MOTOR_BUSY\fP, \fBLABEL_MOTOR_ERROR_INITIALIZATION\fP, \fBLABEL_MOTOR_ERROR_UNEXPECTED_STATUS\fP, \fBLABEL_MOTOR_ERROR_LIMIT_SWITCH\fP, \fBLABEL_MOTOR_ERROR_BRAKE_DEVICE\fP, \fBLABEL_MOTOR_ERROR_TIMOUT\fP, \fBLABEL_MOTOR_ERROR_INTERNAL_FAULT\fP, \fBLABEL_MOTOR_ERROR_ACCESS_REGISTER\fP, \fBLABEL_MOTOR_ERROR_MISSING_HOME\fP, \fBLABEL_MOTOR_ERROR_COMMAND_DISABLED\fP, \fBLABEL_MOTOR_ERROR_COMMAND_ABORTED\fP, \fBLABEL_COMPRESSION_ACTIVATED\fP, \fBLABEL_ARM_ACTIVATED\fP, \fBLABEL_BODY_ACTIVATED\fP, \fBLABEL_VERTICAL_ACTIVATED\fP, \fBLABEL_SLIDE_ACTIVATED\fP, \fBLABEL_TILT_ACTIVATED\fP, \fBNUM_MESSAGES\fP }"
.br
.in -1c
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBNotify\fP ()"
.br
.in -1c
.SS "Static Public Member Functions"

.in +1c
.ti -1c
.RI "static bool \fBsetLanguage\fP (System::String^ language)"
.br
.ti -1c
.RI "static System::String ^ \fBTranslateTitle\fP (\fBmessages\fP msg)"
.br
.ti -1c
.RI "static System::String ^ \fBTranslateContent\fP (\fBmessages\fP msg)"
.br
.ti -1c
.RI "static System::String ^ \fBTranslateNumber\fP (\fBmessages\fP msg)"
.br
.ti -1c
.RI "static System::String ^ \fBTranslateLabel\fP (\fBmessages\fP msg)"
.br
.ti -1c
.RI "static \fBtranslate\fP ^ \fBTranslate\fP (\fBmessages\fP msg)"
.br
.ti -1c
.RI "static bool \fBisError\fP (void)"
.br
.ti -1c
.RI "static bool \fBisWarning\fP (void)"
.br
.ti -1c
.RI "static bool \fBisInfo\fP (void)"
.br
.ti -1c
.RI "static bool \fBisInstant\fP (void)"
.br
.ti -1c
.RI "static void \fBclrInstant\fP (void)"
.br
.ti -1c
.RI "static System::String ^ \fBgetListOfErrors\fP (void)"
.br
.ti -1c
.RI "static System::String ^ \fBformatMsg\fP (\fBmessages\fP msg)"
.br
.ti -1c
.RI "static void \fBdisable\fP (\fBmessages\fP msg)"
.br
.ti -1c
.RI "static void \fBactivate\fP (\fBmessages\fP msg)"
.br
.ti -1c
.RI "static void \fBactivate\fP (\fBmessages\fP msg, System::String^ extra)"
.br
.ti -1c
.RI "static void \fBdeactivate\fP (\fBmessages\fP msg)"
.br
.ti -1c
.RI "static void \fBinstant\fP (\fBmessages\fP msg)"
.br
.ti -1c
.RI "static void \fBinstant\fP (\fBmessages\fP msg, System::String^ extra)"
.br
.ti -1c
.RI "static \fBmessages\fP \fBgetInstant\fP (void)"
.br
.ti -1c
.RI "static void \fBopen_error\fP (Form^ parent)"
.br
.ti -1c
.RI "static void \fBclose_error\fP (void)"
.br
.ti -1c
.RI "static void \fBopen_instant\fP (Form^ parent)"
.br
.ti -1c
.RI "static void \fBclose_instant\fP (void)"
.br
.ti -1c
.RI "static bool \fBisInstantOpen\fP (void)"
.br
.ti -1c
.RI "static bool \fBisErrorOpen\fP (void)"
.br
.in -1c
.SS "Static Public Attributes"

.in +1c
.ti -1c
.RI "static List< \fBtranslate\fP^> ^ \fBdictionary\fP = nullptr"
.br
.ti -1c
.RI "static \fBNotify\fP ^ \fBdevice\fP = gcnew \fBNotify\fP()"
.br
.in -1c
.SS "Static Private Attributes"

.in +1c
.ti -1c
.RI "static List< \fBitem\fP^> ^ \fBmessage_list\fP"
.br
.ti -1c
.RI "static \fBmessages\fP \fBinstant_msg\fP = \fBmessages::NO_MESSAGE\fP"
.br
.ti -1c
.RI "static System::String ^ \fBinstant_extra\fP"
.br
.ti -1c
.RI "static int \fBerror_counter\fP = 0"
.br
.ti -1c
.RI "static int \fBwarning_counter\fP = 0"
.br
.ti -1c
.RI "static int \fBinfo_counter\fP = 0"
.br
.ti -1c
.RI "static \fBErrorForm\fP ^ \fBerrorWindow\fP"
.br
.ti -1c
.RI "static \fBinfoPopupForm\fP ^ \fBinstantWindow\fP"
.br
.in -1c
.SH "Detailed Description"
.PP 
This is the calss implementing the application message management\&. 
.SH "Member Enumeration Documentation"
.PP 
.SS "enum class \fBNotify::messages\fP\fR [strong]\fP"

.PP
\fBEnumerator\fP
.in +1c
.TP
\f(BINO_MESSAGE \fP(0)
.TP
\f(BIERROR_FATAL_COMPILATION \fP
.TP
\f(BIERROR_AWS_INITIALIZATION \fP
.TP
\f(BIERROR_PCB301_RESET \fP
.TP
\f(BIERROR_PCB302_RESET \fP
.TP
\f(BIERROR_PCB303_RESET \fP
.TP
\f(BIERROR_PCB304_RESET \fP
.TP
\f(BIERROR_PCB325_RESET \fP
.TP
\f(BIERROR_PCB326_RESET \fP
.TP
\f(BIERROR_PCB301_COMMUNICATION_ERROR \fP
.TP
\f(BIERROR_PCB302_COMMUNICATION_ERROR \fP
.TP
\f(BIERROR_PCB303_COMMUNICATION_ERROR \fP
.TP
\f(BIERROR_PCB304_COMMUNICATION_ERROR \fP
.TP
\f(BIERROR_PCB325_COMMUNICATION_ERROR \fP
.TP
\f(BIERROR_PCB326_COMMUNICATION_ERROR \fP
.TP
\f(BIERROR_XRAY_PUSH_FAULT \fP
.TP
\f(BIERROR_MOTOR_PEDALS_UD_FAULT \fP
.TP
\f(BIERROR_MOTOR_PEDALS_COMP_FAULT \fP
.TP
\f(BIERROR_MOTOR_BUTTON_CARM_FAULT \fP
.TP
\f(BIERROR_MOTOR_BUTTON_UD_FAULT \fP
.TP
\f(BIERROR_CONFIGURATION_FAULT \fP
.TP
\f(BIERROR_POWER_DOWN_ERROR \fP
.TP
\f(BIERROR_EMERGENCY_BUTTON \fP
.TP
\f(BIERROR_CABINET_SAFETY \fP
.TP
\f(BIERROR_BATTERY_LOW_ERROR \fP
.TP
\f(BIERROR_GENERATOR_ERROR_CONNECTION \fP
.TP
\f(BIERROR_COLLIMATION_SELECTION_ERROR \fP
.TP
\f(BIERROR_FILTER_SELECTION_ERROR \fP
.TP
\f(BIERROR_MIRROR_SELECTION_ERROR \fP
.TP
\f(BIERROR_STATOR_SENSOR_LOW \fP
.TP
\f(BIERROR_STATOR_SENSOR_SHORT \fP
.TP
\f(BIERROR_BULB_SENSOR_LOW \fP
.TP
\f(BIERROR_BULB_SENSOR_SHORT \fP
.TP
\f(BIERROR_BODY_MOTOR_BRAKE_FAULT \fP
.TP
\f(BIERROR_BODY_POTENTIOMETER_READ_FAULT \fP
.TP
\f(BIERROR_TILT_MOTOR_BRAKE_FAULT \fP
.TP
\f(BIERROR_TILT_MOTOR_HOMING \fP
.TP
\f(BIERROR_VERTICAL_MOTOR_HOMING \fP
.TP
\f(BIERROR_VERTICAL_LIMIT_SWITCH \fP
.TP
\f(BIERROR_ARM_MOTOR_HOMING \fP
.TP
\f(BIERROR_SLIDE_MOTOR_HOMING \fP
.TP
\f(BIERROR_BODY_MOTOR_HOMING \fP
.TP
\f(BIERROR_BODY_LIMIT_SWITCH \fP
.TP
\f(BIERROR_GENERATOR_SETUP \fP
.TP
\f(BIERROR_GENERATOR_SYS_MESSAGE \fP
.TP
\f(BIERROR_POTTER_GRID_POSITION \fP
.TP
\f(BIERROR_X_RAY_BUTTON_RELEASED \fP
.TP
\f(BIERROR_X_RAY_LOW_ANODIC \fP
.TP
\f(BIERROR_X_RAY_LOW_KV \fP
.TP
\f(BIERROR_X_RAY_TIMEOUT \fP
.TP
\f(BIERROR_X_RAY_STARTER \fP
.TP
\f(BIWARNING_MESSAGES_SECTION \fP
.TP
\f(BIWARNING_MISSING_COMPRESSION \fP
.TP
\f(BIWARNING_MISSING_PATIENT_PROTECTION \fP
.TP
\f(BIWARNING_ARM_POSITION_WARNING \fP
.TP
\f(BIWARNING_POSITION_NOT_CALIBRATED \fP
.TP
\f(BIWARNING_FORCE_NOT_CALIBRATED \fP
.TP
\f(BIWARNING_MISSING_PADDLE \fP
.TP
\f(BIWARNING_WRONG_PADDLE \fP
.TP
\f(BIWARNING_MISSING_EXPOSURE_MODE \fP
.TP
\f(BIWARNING_MISSING_EXPOSURE_DATA \fP
.TP
\f(BIWARNING_XRAY_BUTTON_DISABLED \fP
.TP
\f(BIWARNING_DOOR_STUDY_OPEN \fP
.TP
\f(BIWARNING_GENERATOR_NOT_READY \fP
.TP
\f(BIWARNING_TUBE_TEMPERATURE \fP
.TP
\f(BIWARNING_COLLI_LIGHT_ACTIVATION \fP
.TP
\f(BIWARNING_STATOR_SENSOR_HIGH \fP
.TP
\f(BIWARNING_BULB_SENSOR_HIGH \fP
.TP
\f(BIWARNING_ANODE_TEMPERATURE_HIGH \fP
.TP
\f(BIWARNING_FILTER_OUT_OF_POSITION \fP
.TP
\f(BIWARNING_MIRROR_OUT_OF_POSITION \fP
.TP
\f(BIWARNING_COLLIMATOR_OUT_OF_POSITION \fP
.TP
\f(BIWARNING_GENERATOR_SERVICE_MODE \fP
.TP
\f(BIWARNING_GRID_OUT_OF_POSITION \fP
.TP
\f(BIWARNING_GRID_GENERAL_ENABLE \fP
.TP
\f(BIINFO_MESSAGES_SECTION \fP
.TP
\f(BIINFO_BATTERY_DISABLED \fP
.TP
\f(BIINFO_PROJECTION_CONFIRMATION \fP
.TP
\f(BIINFO_PROJECTION_ABORT \fP
.TP
\f(BIINFO_SLIDE_ACTIVATION_CONFIRMATION \fP
.TP
\f(BIINFO_BODY_ACTIVATION_CONFIRMATION \fP
.TP
\f(BIINFO_ARM_ACTIVATION_CONFIRMATION \fP
.TP
\f(BIINFO_TILT_ACTIVATION_CONFIRMATION \fP
.TP
\f(BIINFO_SLIDE_ACTIVATION_FAILED \fP
.TP
\f(BIINFO_BODY_ACTIVATION_FAILED \fP
.TP
\f(BIINFO_ARM_ACTIVATION_FAILED \fP
.TP
\f(BIINFO_TILT_ACTIVATION_FAILED \fP
.TP
\f(BIINFO_BODY_DRIVER \fP
.TP
\f(BIINFO_VERTICAL_DRIVER \fP
.TP
\f(BIINFO_TILT_DRIVER \fP
.TP
\f(BIINFO_ARM_DRIVER \fP
.TP
\f(BIINFO_SLIDE_DRIVER \fP
.TP
\f(BIINFO_MOTOR_48V_SAFETY_LINE_OFF \fP
.TP
\f(BIINFO_MOTOR_48V_POWER_SUPPLY_OFF \fP
.TP
\f(BIINFO_BURNING_JUMPER_PRESENT \fP
.TP
\f(BIINFO_POWER_LOCK \fP
.TP
\f(BIINFO_ACTIVATION_MOTOR_MANUAL_DISABLE \fP
.TP
\f(BIINFO_ACTIVATION_MOTOR_SAFETY_DISABLE \fP
.TP
\f(BIINFO_ACTIVATION_MOTOR_ERROR_DISABLE \fP
.TP
\f(BIINFO_POWER_OFF_REQUEST_ACTIVATION \fP
.TP
\f(BIINFO_GRID_INOUT_MANUAL_MODE \fP
.TP
\f(BILABEL_MESSAGES_SECTION \fP
.TP
\f(BILABEL_ERROR_WINDOW_PANEL \fP
.TP
\f(BILABEL_PROJECTION_SELECTION_PANEL \fP
.TP
\f(BILABEL_ZERO_SETTING_PANEL_TITLE \fP
.TP
\f(BILABEL_ROTATION_TOOL_PANEL_TITLE \fP
.TP
\f(BILABEL_SERVICE_PANEL_TITLE \fP
.TP
\f(BILABEL_CALIBRATION_PANEL_TITLE \fP
.TP
\f(BILABEL_CALIBRATION_XRAY_TUBE_TITLE \fP
.TP
\f(BILABEL_NOT_READY_FOR_EXPOSURE \fP
.TP
\f(BILABEL_READY_FOR_EXPOSURE \fP
.TP
\f(BILABEL_X_RAY_ON \fP
.TP
\f(BILABEL_MOTOR_COMMAND_SUCCESS \fP
.TP
\f(BILABEL_MOTOR_COMMAND_MANUAL_TERMINATION \fP
.TP
\f(BILABEL_MOTOR_ERROR_OBSTACLE_DETECTED \fP
.TP
\f(BILABEL_MOTOR_ERROR_MOTOR_BUSY \fP
.TP
\f(BILABEL_MOTOR_ERROR_INITIALIZATION \fP
.TP
\f(BILABEL_MOTOR_ERROR_UNEXPECTED_STATUS \fP
.TP
\f(BILABEL_MOTOR_ERROR_LIMIT_SWITCH \fP
.TP
\f(BILABEL_MOTOR_ERROR_BRAKE_DEVICE \fP
.TP
\f(BILABEL_MOTOR_ERROR_TIMOUT \fP
.TP
\f(BILABEL_MOTOR_ERROR_INTERNAL_FAULT \fP
.TP
\f(BILABEL_MOTOR_ERROR_ACCESS_REGISTER \fP
.TP
\f(BILABEL_MOTOR_ERROR_MISSING_HOME \fP
.TP
\f(BILABEL_MOTOR_ERROR_COMMAND_DISABLED \fP
.TP
\f(BILABEL_MOTOR_ERROR_COMMAND_ABORTED \fP
.TP
\f(BILABEL_COMPRESSION_ACTIVATED \fP
.TP
\f(BILABEL_ARM_ACTIVATED \fP
.TP
\f(BILABEL_BODY_ACTIVATED \fP
.TP
\f(BILABEL_VERTICAL_ACTIVATED \fP
.TP
\f(BILABEL_SLIDE_ACTIVATED \fP
.TP
\f(BILABEL_TILT_ACTIVATED \fP
.TP
\f(BINUM_MESSAGES \fP
.SH "Constructor & Destructor Documentation"
.PP 
.SS "Notify::Notify ()\fR [inline]\fP"

.SH "Member Function Documentation"
.PP 
.SS "void Notify::activate (\fBmessages\fP msg)\fR [static]\fP"

.SS "void Notify::activate (\fBmessages\fP msg, System::String^ extra)\fR [static]\fP"

.SS "static void Notify::close_error (void )\fR [inline]\fP, \fR [static]\fP"

.SS "static void Notify::close_instant (void )\fR [inline]\fP, \fR [static]\fP"

.SS "static void Notify::clrInstant (void )\fR [inline]\fP, \fR [static]\fP"

.SS "void Notify::deactivate (\fBmessages\fP msg)\fR [static]\fP"

.SS "void Notify::disable (\fBmessages\fP msg)\fR [static]\fP"

.SS "System::String Notify::formatMsg (\fBmessages\fP msg)\fR [static]\fP"

.SS "static \fBmessages\fP Notify::getInstant (void )\fR [inline]\fP, \fR [static]\fP"

.SS "System::String Notify::getListOfErrors (void )\fR [static]\fP"

.SS "void Notify::instant (\fBmessages\fP msg)\fR [static]\fP"

.SS "void Notify::instant (\fBmessages\fP msg, System::String^ extra)\fR [static]\fP"

.SS "static bool Notify::isError (void )\fR [inline]\fP, \fR [static]\fP"

.SS "static bool Notify::isErrorOpen (void )\fR [inline]\fP, \fR [static]\fP"

.SS "static bool Notify::isInfo (void )\fR [inline]\fP, \fR [static]\fP"

.SS "static bool Notify::isInstant (void )\fR [inline]\fP, \fR [static]\fP"

.SS "static bool Notify::isInstantOpen (void )\fR [inline]\fP, \fR [static]\fP"

.SS "static bool Notify::isWarning (void )\fR [inline]\fP, \fR [static]\fP"

.SS "static void Notify::open_error (Form^ parent)\fR [inline]\fP, \fR [static]\fP"

.SS "static void Notify::open_instant (Form^ parent)\fR [inline]\fP, \fR [static]\fP"

.SS "bool Notify::setLanguage (System::String^ language)\fR [static]\fP"

.SS "\fBNotify::translate\fP Notify::Translate (\fBmessages\fP msg)\fR [static]\fP"

.SS "System::String Notify::TranslateContent (\fBmessages\fP msg)\fR [static]\fP"

.SS "static System::String ^ Notify::TranslateLabel (\fBmessages\fP msg)\fR [inline]\fP, \fR [static]\fP"

.SS "System::String Notify::TranslateNumber (\fBmessages\fP msg)\fR [static]\fP"

.SS "System::String Notify::TranslateTitle (\fBmessages\fP msg)\fR [static]\fP"

.SH "Member Data Documentation"
.PP 
.SS "\fBNotify\fP ^ Notify::device = gcnew \fBNotify\fP()\fR [static]\fP"

.SS "List<\fBtranslate\fP^> ^ Notify::dictionary = nullptr\fR [static]\fP"

.SS "int Notify::error_counter = 0\fR [static]\fP, \fR [private]\fP"

.SS "\fBErrorForm\fP ^ Notify::errorWindow\fR [static]\fP, \fR [private]\fP"

.SS "int Notify::info_counter = 0\fR [static]\fP, \fR [private]\fP"

.SS "System::String ^ Notify::instant_extra\fR [static]\fP, \fR [private]\fP"

.SS "\fBmessages\fP Notify::instant_msg = \fBmessages::NO_MESSAGE\fP\fR [static]\fP, \fR [private]\fP"

.SS "\fBinfoPopupForm\fP ^ Notify::instantWindow\fR [static]\fP, \fR [private]\fP"

.SS "List<\fBitem\fP^> ^ Notify::message_list\fR [static]\fP, \fR [private]\fP"

.SS "int Notify::warning_counter = 0\fR [static]\fP, \fR [private]\fP"


.SH "Author"
.PP 
Generated automatically by Doxygen for MCPU from the source code\&.


PROJECT_MCPU/MCPU/MCPU_MASTER/MESSAGES/Notify.cpp
#include "Notify.h"
#include "mutex"

static std::mutex gantry_errors_mutex;

typedef struct {	
	const char* error;
	Notify::messages msg;
	const char* title;
	const char* content;
} language_item;

static const language_item ENG_ARRAY[] = {
	#include "translate_eng.h"
	{"LAST",Notify::messages::NO_MESSAGE,"",""}
};

static const language_item ITA_ARRAY[] = {
	#include "translate_ita.h"
	{"LAST",Notify::messages::NO_MESSAGE,"",""}
};

static const language_item FRA_ARRAY[] = {
	#include "translate_fra.h"
	{"LAST",Notify::messages::NO_MESSAGE,"",""}
};

static const language_item ESP_ARRAY[] = {
	#include "translate_esp.h"
	{"LAST",Notify::messages::NO_MESSAGE,"",""}
};

static const language_item POR_ARRAY[] = {
	#include "translate_por.h"
	{"LAST",Notify::messages::NO_MESSAGE,"",""}
};

static const language_item RUS_ARRAY[] = {
	#include "translate_rus.h"
	{"LAST",Notify::messages::NO_MESSAGE,"",""}
};

static const language_item DEFAULT_ARRAY[] = {
	{"LAST",Notify::messages::NO_MESSAGE,"",""}
};

bool Notify::setLanguage(System::String^ language) {

	const language_item* pLanguage;
	bool language_ok = true;
	if (language == "ITA")  pLanguage = ITA_ARRAY;
	else if (language == "ENG")  pLanguage = ENG_ARRAY;
	else if (language == "FRA")  pLanguage = FRA_ARRAY;
	else if (language == "ESP")  pLanguage = ESP_ARRAY;
	else if (language == "POR")  pLanguage = POR_ARRAY;
	else if (language == "RUS")  pLanguage = RUS_ARRAY;
	else {
		pLanguage = DEFAULT_ARRAY;
		language_ok = false;
	}

	dictionary = gcnew List<translate^>();
	
	int i = 0;
	while (1) {
		if (pLanguage[i].msg == messages::NO_MESSAGE) break;
		System::String^ err = gcnew System::String(pLanguage[i].error);
		System::String^ title = gcnew System::String(pLanguage[i].title);
		System::String^ content = gcnew System::String(pLanguage[i].content);

		dictionary->Add(gcnew translate(err, pLanguage[i].msg, title, content));
		i ++;
	}

	return language_ok;

};

System::String^ Notify::TranslateTitle(messages msg) {
	for (int i = 0; i < dictionary->Count; i++)
	{
		if (dictionary[i]->msg == msg) return dictionary[i]->title;
	}
	return "";	
}

System::String^ Notify::TranslateContent(messages msg) {
	for (int i = 0; i < dictionary->Count; i++)
	{
		if (dictionary[i]->msg == msg) return dictionary[i]->content;
	}
	return "";
}

System::String^ Notify::TranslateNumber(messages msg) {
	for (int i = 0; i < dictionary->Count; i++)
	{
		if (dictionary[i]->msg == msg) return dictionary[i]->number;
	}
	return "";
}


Notify::translate^ Notify::Translate(messages msg) {
	for (int i = 0; i < dictionary->Count; i++)
	{
		if (dictionary[i]->msg == msg) return dictionary[i];
	}
	return nullptr;
}


void Notify::activate(messages msg) {
	const std::lock_guard<std::mutex> lock(gantry_errors_mutex);

	if (msg >= messages::LABEL_MESSAGES_SECTION) return;
	if (msg == messages::NO_MESSAGE) return;
	if (message_list[(int) msg]->disabled) return;
	if(message_list[(int)msg]->active) return;

	message_list[(int)msg]->active = true;	
	message_list[(int)msg]->extra = nullptr;
	
	
	if (msg < messages::WARNING_MESSAGES_SECTION) error_counter++;
	else if (msg < messages::INFO_MESSAGES_SECTION) warning_counter++;
	else  info_counter++;

}

void Notify::instant(messages msg) {
	const std::lock_guard<std::mutex> lock(gantry_errors_mutex);

	if (msg >= messages::LABEL_MESSAGES_SECTION) return;
	if (msg == messages::NO_MESSAGE) return;
	if (message_list[(int)msg]->disabled) return;
	
	instant_msg = msg;
	instant_extra = nullptr;
	return;

}

void Notify::activate(messages msg, System::String^ extra) {
	const std::lock_guard<std::mutex> lock(gantry_errors_mutex);

	if (msg >= messages::LABEL_MESSAGES_SECTION) return;
	if (msg == messages::NO_MESSAGE) return;
	if (message_list[(int)msg]->disabled) return;
	if (message_list[(int)msg]->active) return;


	message_list[(int)msg]->active = true;
	message_list[(int)msg]->extra = extra;
	
	if (msg < messages::WARNING_MESSAGES_SECTION) error_counter++;
	else if (msg < messages::INFO_MESSAGES_SECTION) warning_counter++;
	else  info_counter++;
	

}

void Notify::instant(messages msg, System::String^ extra) {
	const std::lock_guard<std::mutex> lock(gantry_errors_mutex);

	if (msg >= messages::LABEL_MESSAGES_SECTION) return;
	if (msg == messages::NO_MESSAGE) return;
	if (message_list[(int)msg]->disabled) return;
	
	instant_msg = msg;
	instant_extra = extra;
	return;
}


void Notify::deactivate(messages msg) {
	const std::lock_guard<std::mutex> lock(gantry_errors_mutex);

	if (msg >= messages::LABEL_MESSAGES_SECTION) return;
	if (msg == messages::NO_MESSAGE) return;
	if (!message_list[(int)msg]->active) return;
	message_list[(int)msg]->active = false;
	message_list[(int)msg]->extra = nullptr;

	if (msg < messages::WARNING_MESSAGES_SECTION) error_counter--;
	else if (msg < messages::INFO_MESSAGES_SECTION) warning_counter--;
	else  info_counter--;

}

void Notify::disable(messages msg) {
	const std::lock_guard<std::mutex> lock(gantry_errors_mutex);
	if (msg >= messages::LABEL_MESSAGES_SECTION) return;
	if (msg == messages::NO_MESSAGE) return;
	message_list[(int)msg]->disabled = true;
	if (!message_list[(int)msg]->active) return;

	message_list[(int)msg]->active = false;
	message_list[(int)msg]->extra = nullptr;

	if (msg < messages::WARNING_MESSAGES_SECTION) error_counter--;
	else if (msg < messages::INFO_MESSAGES_SECTION) warning_counter--;
	else  info_counter--;

}


System::String^ Notify::formatMsg(messages msg) {
	translate^ msgit = Translate(msg);
	if (msgit == nullptr) return "";
	return "[" + msgit->number + "]: " + msgit->title + "\n" + msgit->content;
}

System::String^ Notify::getListOfErrors(void) {
	const std::lock_guard<std::mutex> lock(gantry_errors_mutex);

	System::String^ errlist = "";
	
	// Sets the full list
	for (int i = (int)messages::NO_MESSAGE + 1; i < (int) messages::NUM_MESSAGES; i++) {		
		if (!message_list[i]->active) continue;
		if (i == (int)messages::WARNING_MESSAGES_SECTION) continue;
		if (i == (int)messages::INFO_MESSAGES_SECTION) continue;

		errlist += formatMsg((messages) i);
		if (message_list[i]->extra) errlist += ":\n >" + message_list[i]->extra;
		errlist += "\n----------------------------------------------------------------------------\n\n";
	}


	return errlist;
}

